### PR TITLE
Port to Yocto scarthgap

### PR DESCRIPTION
**Pull request by quaresmajose**
It's not ready yet, but there's not much left and it's already at a point where it can start to be reviewed.

The commits that start with [FIXME]/ [WIP] still need to be worked on

-  [FIXME] bsp: lmp-machine-custom: drop recovery
-  [FIXME] bsp: lmp-machine-custom: imx6ullevk: drop KERNEL_DEVICETREE
-  [FIXME] base: systemd: drop patch
-  [FIXME] base: optee-os-fio: drop patch
-  [WIP] base: docker-compose: drop patch
-  [WIP] base: docker-ce -> docker-moby

### CODE REVIEW

**Comment by ricardosalveti** [comment]
We should probably break this pr in smaller ones, otherwise it makes it impossible to review and merge. We can separate the fixme patches as well.

Not concerned to have main in a broken / partial state at this stage, until we land everything we need.

File: meta-lmp-bsp/recipes-bsp/trusted-firmware-a/tf-a-fio_2.7.bbappend [hunk 1]
```diff
@@ -17,3 +17,5 @@ SRC_URI:append:stm32mp1common = " \
 SRC_URI:append:sota:stm32mp1common = " \
      file://0003-FIO-interal-stm32mp1-support-for-boot-script-in-FIP.patch \
      "
+
+SRC_URI:append:stm32mp1common = " file://0001-Binutils-2.39-now-warns-when-a-segment-has-RXW-permi.patch"
```
Comment (angolini) [inline meta-lmp-bsp/recipes-bsp/trusted-firmware-a/tf-a-fio_2.7.bbappend:21]: Why this is only for stm32mp1common?

File: meta-lmp-base/recipes-core/ovmf/ovmf_git.bbappend [hunk 2]
```diff
@@ -1,9 +1,2 @@
-FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"
```
Comment (angolini) [inline meta-lmp-base/recipes-core/ovmf/ovmf_git.bbappend:1]: there is a typo on this commit log

pacthes/patches

File: meta-lmp-bsp/recipes-bsp/trusted-firmware-a/tf-a-fio_2.7.bbappend (hunk 1 above)
Comment (quaresmajose) [reply meta-lmp-bsp/recipes-bsp/trusted-firmware-a/tf-a-fio_2.7.bbappend:21]: good question, maybe is not need anymore. I will check locally

File: meta-lmp-base/recipes-core/ovmf/ovmf_git.bbappend (hunk 2 above)
Comment (quaresmajose) [reply meta-lmp-base/recipes-core/ovmf/ovmf_git.bbappend:1]: thanks, I will fix it

**Comment by quaresmajose** [comment]
> We should probably break this pr in smaller ones, otherwise it makes it impossible to review and merge. We can separate the fixme patches as well.
> 
> Not concerned to have main in a broken / partial state at this stage, until we land everything we need.

I'm going to try to break it into some pieces but in a way that doesn't break the main branch.

**Comment by quaresmajose** [comment]
I will propose a new version soon in smaller pieces

**Pull request closed without merging**